### DISCUSSION TITLE
DOCUMENTATION: The software should be traceable and observable.

### DISCUSSION THREAD

**Discussion by martasp** [hassanhabib/The-Standard#219] [General]
There should be a chapter about software engineering **traceability** and **observability** topics.

So we have Traces, Metrics, Logs and we have static metadata about our software engineering like issues, stories, epics, bugs, and customer complaints, etc.  So the dream is to enrich these types of data by combining them into one traceable graph of data. 

Sources of what I'm talking about : 
https://opentelemetry.io/ - Traces, Metrics, Logs
https://github.com/backstage/backstage - Metadata of services and API'S.
https://en.wikipedia.org/wiki/Requirements_traceability - Every requirement or story should be back traceable to code
https://zipkin.io/ - Distributed tracing 
https://en.wikipedia.org/wiki/Static_program_analysis - Code metadata
https://en.wikipedia.org/wiki/Causal_graph - by having a causality graph you can rigorously answer questions.

**Comment by hassanhabib** [comment]
Coming soon.
